staging/src/k8s.io/pod-security-admission/admission/admission.go

```diff
@@ -437,21 +443,44 @@ func (a *Admission) EvaluatePodsInNamespace(ctx context.Context, namespace strin
 	for i, pod := range pods {
 		// short-circuit on exempt runtimeclass
 		if a.exemptRuntimeClass(pod.Spec.RuntimeClassName) {
+			if i == len(pods)-1 {
+				warnings = aggregateWarnings(podsWarings, warnings)
+			}
 			continue
 		}
-		r := policy.AggregateCheckResults(a.Evaluator.EvaluatePod(enforce, &pod.ObjectMeta, &pod.Spec))
+		r, reasons := policy.AggregateCheckResults(a.Evaluator.EvaluatePod(enforce, &pod.ObjectMeta, &pod.Spec))
```

Comment:
`reasons` is part of `r`.
```
reasons := r.ForbiddenReasons
```
A little confused about the func refactor of `AggregateCheckResults`.

Reply:
Thanks for pointing out, there is indeed no need to refactor `AggregateCheckResults`.